### DISCUSSION TITLE
How to influence key of (root) tab screen

### DISCUSSION THREAD

**Discussion by LouiseBC** [react-navigation/react-navigation#10510] [Q&A] [answered]
Given the following configuration:
```
    <Tab.Navigator>
      <Tab.Screen
        name="Home"
        component={Home}
        navigationKey={'home-root'}
      />
    </Tab.Navigator>
```

I would expect the `key` of this screen to be 'home-root', but if you log the navigation state routes, you see something like this:
```
"routes": [{"key": "Home-gWgY508NymlNNN9IXFIMt", "name": "Home", "params": [Object]}]
```

Is it possible to influence the key for the tab root? Or if not, what does this prop do?
I was hoping to be able to set the `navigationKey` prop so I can navigate to this screen using `navigation.navigate({name: 'Home', key: 'home-root', ..})`

**Comment by raajnadar** [comment, marked as answer]
You can use this prop https://reactnavigation.org/docs/screen/#getid

Navigation with the key will be deprecated in the next major release https://github.com/react-navigation/react-navigation/discussions/10506

**Reply by LouiseBC** [reply]
Thanks! This solves my problem, but it seems like the `getId` prop doesn't actually influence the `key`, right? When I log the navigation state the `key` is still generated, and not what I specified. Like I said it doesn't matter in my case because, I'm just asking for completeness

**Reply by satya164** [reply]
The route key is an internal implementation detail and you can't customize it (unless you use a custom router and navigator).

**Reply by satya164** [reply]
Regarding `navigationKey`: https://reactnavigation.org/docs/screen/#navigationkey

**Comment by mahi8813** [comment]
@satya164 getId for a Tab.screen is not working in 5.x.x versions. But docs says getId support is there. How to make it work in 5.x.x.
Support was added in this commit https://github.com/react-navigation/react-navigation/commit/7430c723ced1101191cfbdf0c26d0b0947602b2b
